_includes/articles/dashboard-writeup.md

```diff
@@ -0,0 +1,101 @@
+
+An [example JSON file with an explanation of added fields](https://github.com/bitcoinops/btc-dashboard/blob/master/STATS_TRACKED.md) is available in the same repository.
+ 
+## Implementation
+The rest of this post will be about my experience building the dashboard. To start, I looked at existing tools for blockchain analysis, like BlockSci. Although BlockSci works pretty well, it has a long setup time and I was also looking to analyze blocks as they were being confirmed. In my experience using BlockSci, I found it very easy to write queries similar to [those in the demo they produced](https://citp.github.io/BlockSci/demo.html), which took just a few minutes to return a result. The main downside with using BlockSci is that its parser takes a long time to pre-process blockchain data, but once that is done it performs quite well.
+
+After exploring these options, I decided I can probably get most of the stats needed for the dashboard through RPCs from bitcoind. I wrote some simple code (using btcd's RPC client) to get the stats I needed from each transaction with the `getblock` and `getrawtransaction` RPCs. This code worked pretty well for smaller blocks, but when I tried using it to get the stats of a more recent block - which often have over a thousand transactions - I quickly noticed a problem. My code was calling `getrawtransaction` for every single transaction input in the block. Asking bitcoind to do thousands of RPCs for every single block quickly becomes unsustainable, and was taking a very long amount of time. 
+
+This code looked something like this:
+```
+block, err := rpcclient.GetBlock(blockHash)
+for _, tx := range block.Transactions {
+    for _, input := range tx.TxIn {
+	    analyzeInput(input)
+            
+		prevTx := rpcclient.GetRawTransaction(input.PreviousOutPoint.Hash)
+		analyzePrevOut(input, prevTx)
+	}
+
+	for _, output := range tx.TxOut {
+			analyzeOutput(output)
+	}
+}
+```
+
+
+I then started looking at the `getblockstats` RPC for some additional stats while I figure out a way around this problem. [`getblockstats` is an RPC in `bitcoind`](https://github.com/bitcoin/bitcoin/pull/10757) that gives several useful stats for any given block. For an idea as to what it tells us about a block, the dashboard uses it to show evolving fee-rates, changes in the size of the UTXO set, and the number of inputs, outputs, and transactions for every block.
```

Comment:
I suggest you remove the backticks around bitcoind, since you haven't styled it `bitcoind` elsewhere in the document.